docs/app/routes/docs/guides/react-three-fiber.mdx

```diff
@@ -0,0 +1,401 @@
+and `react-three-fiber` you'll need to install the `@react-spring/three` package.
+
+```bash
+yarn add @react-spring/three
+```
+
+## Why use the library?
+
+A common question asked is why use `react-spring` with `react-three-fiber` when you can use the `useFrame` hook to update your meshes & objects
+instead without knowing another API. Well this is a great question, lets consider a simple use-case.
+
+I have a distortion blob and I want to it to change color on click. You could do this with `useFrame` to perform frame by frame updates, `useRef`
+to access the material object and use the `THREE.Color.lerp` function, slowly incrementing by lerp value until the color is reached. But this then
+requires that I keep track of a `THREE` instance of `Color` which _can_ lead to memory leaks if not handled correctly and not only that, but it would
+be a lot of code. Then you'd need to think about writing your own easing functions. With `react-spring` you can do this in a few lines of code.
+
+```tsx live=true template=r3f showLineNumbers lines=9,12,20 defaultOpen=true
+import { useState } from 'react'
+import { useSpring, animated } from '@react-spring/three'
+import { MeshDistortMaterial } from '@react-three/drei'
+import { Canvas } from '@react-three/fiber'
+
+const AnimatedMeshDistortMaterial = animated(MeshDistortMaterial)
+
+const MyScene = () => {
+  const [clicked, setClicked] = useState(false)
+
+  const springs = useSpring({
+    color: clicked ? '#569AFF' : '#ff6d6d',
+  })
+
+  const handleClick = () => setClicked(s => !s)
+
+  return (
+    <mesh onClick={handleClick}>
+      <sphereGeometry args={[1.5, 64, 32]} />
+      <AnimatedMeshDistortMaterial
+        speed={5}
+        distort={0.5}
+        color={springs.color}
+      />
+    </mesh>
+  )
+}
+
+export default function MyComponent() {
+  return (
+    <Canvas>
+      <ambientLight intensity={0.8} />
+      <pointLight intensity={1} position={[0, 6, 0]} />
+      <MyScene />
+    </Canvas>
+  )
+}
+```
+
+## Use the imperative API
+
+In the example below, we use the power of the `react-spring`'s api to lean into the imperative requirements
+of working with performant webGL scenes. The blob follows you round the canvas (lines 70-82) & scales on
+interaction (lines 46-56) without a single react render to cause these updates.
+
+In addition we use the provide a function to the `config` prop instead of an object to deliver a more sticky
+config for the spring in general, but to give the blob a bouncy feel when the `scale` key is changed – lines 18-30.
+
+Finally, because we're using the position of the mouse which can be considered a `Vector2` and the `position`
+of a `mesh` is a `Vector3` we use a custom interpolation via the `to` method of a `SpringValue` to interpolate
+the array, this can be seen on line 97.
+
+Why not read more about it the [imperative](/docs/concepts/imperative-api) and [interpolation](/docs/advanced/interpolation) APIs?
+
```

Comment:
You might want to link related readings with a subtle aside or note. In readmes like [pmndrs/react-ogl](https://github.com/pmndrs/react-ogl) I'd say something like (also see ...) so it doesn't break from the content flow.